### DISCUSSION TITLE
[Design] Abductor rework

### DISCUSSION THREAD

**Discussion by datlo** [ParadiseSS13/Paradise#18516] [Documents]
Abductors are currently relatively fun for the abductor team but provides almost no counterplay for security to fight them. 
The aim of this rework is to give abductors more tools in their arsenal at the cost of removing their free teleporting out ability, allowing sec a chance to actually catch and kill abductors. Also make abductors fit with the removal of instant stun based combat.

Details of these changes I'm still figuring out, but here's an outline of them : 

* Remove easy, free, instantaneous get out of jail free cards for abductors. An abductor caught in the open, while not an immediate death sentence, should come at a cost to the abductor team. This comes in the form of entry and exit points for the abductor to enter and leave the station through, while having access to an emergency teleport that costs permanent UFO ressource to use, but can be restored using credits gained from abductions.

This change means that if security catches you often enough, eventually, you will not have a way to escape and you will die, as you should for your failure.

This emergency teleport is now triggered by the abductor on station wishing to escape and not their partner, and can be activated no matter what state you are in. It can also be remotely triggered by your partner should you die on station.

* Decouple the abductor console from the station's camera network. Abductors now have oversight on the entire station. Prevent 'cutting cameras' being the best option to prevent abductions.

* Add an advanced abductor cryo pod that fixes any and all forms of damage, slowly overtime. Can also be used to revive/'clone' dead people at the cost of UFO ressource, assuming you've retrieved the body. This prevents the need to steal station medical supplies, which I've always found doesnt fit the abductor's theme.

* Streamline the abductor baton to only have a single mode that does everything depending on the target's state : Knockdown, then stun, then sleep, then cuff. Fits with the new stun mechanics we are moving to. Possibly give the baton nodrop or at least disarm immunity when used by an abductor to prevent lucky disarm and theft by unarmed civilians. 

* Remove strict roles for scientist and agent. While the two players will start with assigned gear and a role, they will be free to swap gear and perform each other's roles if they wish. This does NOT remove the need to cooperate, and while it should be possible to solo abduct in case your partner dies and you are out of ressources to revive them, it will be much more difficult to do so.

* Remove or rework the scientist's science tool and streamline the scientist's role to be easier for new players, while increasing space for veteran players to improve and master the role. Also remove the need for the scientist to teleport down to the station to abduct : The agent will instead bring abductees to exit points for remote retrieval.

* Add additional, balanced mobility tool(s) to the agent to allow evading Security without having to spend ressources on an emergency teleport, while still giving Security a chance to do their thing.

* Add options for the scientist to interact with the station remotely at the cost of temporary, recharging ressources : Force open an airlock and bolt it, deploy forcefields to block security's passage, and more I still need to figure out and balance. This is restricted in area, only working when near an abductor agent, to prevent random grief.

* Add an abductor hud implant which both abductors start with and can also be printed by RND if they salvage and deconstruct said implant for maximum abductor tech. This implant provides both health and diagnostic huds, and also a unique abductor hud that shows : Abductor scientist camera, AI camera (with different color), entry and exit points, and any other otherwise invisible abductor related mechanics.

* Refactor abductor code to be datumized.

* Port tg abductor changes/tools that fits within this new design and are interesting.

I am fairly busy these days, but I love abductors and I want to see the antag shine, so I will likely find some time to code this when I've figured out all the details.

Feel free to DM or ping me on discord at Pi#3723 if you have feedback or suggestions for this rework.

**Comment by BAGELMENSK** [comment]
cloaking would be a cool tech-based way to avoid sec
